x-pack/plugins/fleet/server/services/package_policy.ts

```diff
@@ -1111,12 +1112,13 @@ export function overridePackageInputs(
     }
 
     if (override.vars) {
-      originalInput = deepMergeVars(originalInput, override) as NewPackagePolicyInput;
+      // deepMerge mutate the input
+      deepMergeVars(originalInput, override) as NewPackagePolicyInput;
```

Comment:
This `deepMergeVars` function shouldn't be mutative anymore. Looking at it below it copies everything into a `result` var, e.g. `const result = { ...original } `, performs the merge on that and returns it at the end.

Reply:
Unfortunately it is, and this code will not work if the code was not mutative as we never reuse `originalInput` later. I can take a deeper look at that and try to make `deepMergeVars` immutable

Reply:
Ah I understand. We don't reuse `originalInput`, but we do mutate it above e.g. `originalInput.enabled = override.enabled`, etc. 

We do mutate the object as it appears in the `inputs` array here, but it doesn't seem like `deepMergeVars` is doing any mutation directly as I understand it. If we wanted to completely avoid mutating the `inputs` array here we'd need to substantially refactor.